core/src/client/async_client/mod.rs

```diff
@@ -162,8 +162,15 @@ impl ClientBuilder {
 	///
 	/// # Panics
 	///
-	/// This function panics if `max` is 0.
+	/// This function panics if `max` is 0 or bigger than usize::MAX / 2.
```

Comment:
This now allocates the entire capacity up front, so I think we should be super clear that this is now allocating when before it wasn't.

I'd suggest changing the name entirely to eg `with_capacity_per_subscription` (partly so that anybody using this function is forced to find the new one) and make it clear in the docs that it's not a "max capacity" but an actual allocating "capacity" (like `Vec::with_capacity`)

Reply:
It's doesn't allocate when the client is instantiated but when each subscription is created.

Sure good idea.